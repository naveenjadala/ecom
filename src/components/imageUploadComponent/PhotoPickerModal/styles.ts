import {StyleSheet} from 'react-native';
import {getFontSize} from '../../../assets/fonts/FontSizes';
import {colors} from '../../../assets/colors';

export const photoPickerStyles = StyleSheet.create({
  container: {
    flex: 1,
    height: '100%',
    width: '100%',
    backgroundColor: 'rgba(52, 52, 52, 0.1)',
  },
  subContainer: {
    position: 'absolute',
    bottom: 0,
    width: '100%',
  },
  text: {
    textAlign: 'center',
    fontSize: getFontSize(18),
    color: colors.text,
  },
  buttonContainer: {
    width: '95%',
    alignSelf: 'center',
    backgroundColor: colors.primary,
    marginBottom: 8,
    borderRadius: 8,
    elevation: 1,
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 1},
    shadowOpacity: 0.2,
    shadowRadius: 1,
  },
  button: {
    padding: 15,
  },
});
