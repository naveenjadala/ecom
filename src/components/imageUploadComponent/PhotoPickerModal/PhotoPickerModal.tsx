import {View, Text, Modal, TouchableOpacity} from 'react-native';
import React, {FC} from 'react';
import {photoPickerStyles} from './styles';
import {Divider} from 'react-native-paper';

type PhotoPickerModalProps = {
  visible: boolean;
  closeModal: () => void;
  openCamera: () => void;
  openGallery: () => void;
};

/**
 * Photo picker modal component.
 *
 * @param {PhotoPickerModalProps} visible - Flag to control visibility
 * @param {function} setVisible - Function to set visibility
 * @return {JSX.Element} The modal component with photo picker options
 */
const PhotoPickerModal: FC<PhotoPickerModalProps> = ({
  visible,
  closeModal,
  openCamera,
  openGallery,
}) => {
  return (
    <Modal
      testID="photoPickerModal"
      visible={visible}
      onRequestClose={closeModal}
      presentationStyle="overFullScreen"
      animationType="slide"
      statusBarTranslucent={false}
      transparent>
      <View style={photoPickerStyles.container}>
        <View style={photoPickerStyles.subContainer}>
          <View style={photoPickerStyles.buttonContainer}>
            <TouchableOpacity
              testID="takePhoto"
              style={photoPickerStyles.button}
              onPress={openCamera}>
              <Text style={photoPickerStyles.text}>Take Photo</Text>
            </TouchableOpacity>
            <Divider />
            <TouchableOpacity
              testID="choosePhoto"
              style={photoPickerStyles.button}
              onPress={openGallery}>
              <Text style={photoPickerStyles.text}>Choose Photo</Text>
            </TouchableOpacity>
          </View>
          <View style={photoPickerStyles.buttonContainer}>
            <TouchableOpacity
              testID="cancel"
              accessibilityRole="button"
              style={photoPickerStyles.button}
              onPress={() => closeModal()}>
              <Text style={photoPickerStyles.text}>Cancel</Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </Modal>
  );
};

export default PhotoPickerModal;
