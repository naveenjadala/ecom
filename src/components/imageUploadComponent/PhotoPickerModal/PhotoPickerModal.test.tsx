import {render, screen} from '@testing-library/react-native';
import React from 'react';
import PhotoPickerModal from './PhotoPickerModal';

describe('PhotoPickerModal', () => {
  it('renders correctly', () => {
    render(
      <PhotoPickerModal
        visible={false}
        closeModal={() => {}}
        openCamera={() => {}}
        openGallery={() => {}}
      />,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});

// describe('PhotoPickerModal', () => {
//   test('should render elements', () => {
//     render(<PhotoPickerModal visible={false} closeModal={() => {}} />);
//     // const getTakePhotoButton = screen.getByTestId('takePhoto');
//     // const getChoosePhotoButton = screen.getByTestId('choosePhoto');
//     const getCancelButton = screen.getByRole('button');
//     // expect(getTakePhotoButton).toBeOnTheScreen();
//     // expect(getChoosePhotoButton).toBeOnTheScreen();
//     expect(getCancelButton).toBeTruthy();
//   });
// });
