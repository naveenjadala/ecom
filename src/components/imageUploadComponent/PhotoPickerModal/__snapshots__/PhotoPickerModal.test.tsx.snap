// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoPickerModal renders correctly 1`] = `null`;
